libs/etterlatte-token-model/src/main/kotlin/no/nav/etterlatte/token/Bruker.kt

```diff
@@ -0,0 +1,49 @@
+package no.nav.etterlatte.token
+
+sealed class Bruker(
+    open val accessToken: String
+) {
+    abstract fun saksbehandlerIdentEllerSystemnavn(): String
+
+    abstract fun erMaskinTilMaskin(): Boolean
+
+    abstract fun saksbehandlerEnhet(saksbehandlere: Map<String, String>): String
+
+    companion object {
+        private fun erMaskinTilMaskin(oid: String?, sub: String?) = (oid == sub) && (oid != null)
+        fun of(accessToken: String, saksbehandler: String?, oid: String?, sub: String?): Bruker {
+            return if (erMaskinTilMaskin(oid = oid, sub = sub)) {
+                System(accessToken, oid!!, sub!!)
+            } else if (saksbehandler != null) {
+                Saksbehandler(accessToken, saksbehandler)
+            } else {
+                throw Exception(
+                    "Er ikke maskin-til-maskin, og Navident er null i token, sannsynligvis manglende claim NAVident"
+                )
+            }
+        }
+    }
+}
+
+data class System(override val accessToken: String, val oid: String, val sub: String) : Bruker(accessToken) {
+    override fun erMaskinTilMaskin() = true
+
+    override fun saksbehandlerIdentEllerSystemnavn() = Fagsaksystem.EY.name
+
+    override fun saksbehandlerEnhet(saksbehandlere: Map<String, String>) = Fagsaksystem.EY.name
```

Comment:
Skurrer litt med at den må ta inn et map her som ikke brukes videre, men ser ikke helt i farta noen annen god måte å gjøre dette på annet enn å ha egen metode for hver sub-klasse og det blir kanskje litt klønete der de brukes?

Reply:
> Skurrer litt med at den må ta inn et map her som ikke brukes videre, men ser ikke helt i farta noen annen god måte å gjøre dette på annet enn å ha egen metode for hver sub-klasse og det blir kanskje litt klønete der de brukes?

Einig, men usikker på om eg har lyst til å gjera noko med denne no, det blir jo uansett endringar i saksbehandlere-opplegget når vi kjem ut av pilot.